Add critical path section for CI Pipelines (#23767)
DataDog/documentation: content/en/continuous_integration/search/_index.md

The unified trace shows in a single trace all pipeline traces generated due to the different partial retries of your pipeline. If the pipeline execution has no partial retries, the unified trace shows only the trace of a single pipeline execution.

### Highlight critical path

To highlight the critical path on the trace, click on the `Critical path` checkbox on the pipeline execution page.

> The critical path highlights the spans speeding which will speed the overall execution time of your pipeline. If a CI job is on the critical path, it means it is part of the longest weighted path through the trace in terms of execution time. Speeding up the CI Jobs on the critical path is strictly necessary to speed up the CI pipeline.

Reviewer: ```suggestion
The critical path highlights the spans that, when shortened, reduce the overall execution time of your pipeline. If a CI job is on the critical path, it means it is part of the longest weighted path through the trace in terms of execution time. Speeding up the CI Jobs on the critical path is strictly necessary to speed up the CI pipeline.
```

> The critical path highlights the spans speeding which will speed the overall execution time of your pipeline. If a CI job is on the critical path, it means it is part of the longest weighted path through the trace in terms of execution time. Speeding up the CI Jobs on the critical path is strictly necessary to speed up the CI pipeline.

Reviewer 2: `is part of the longest weighted path`

Isn't it just the longest? Do we add any weights, and if so which?
